addons/html_builder/static/tests/snippets_menu.test.js

```diff
@@ -54,3 +92,104 @@ test("drag and drop snippet inner content", async () => {
+            ],
+        },
+    });
+    await openSnippetsMenu();
+    const snippetGroupsSelector = `.o-snippets-menu [data-category="snippet_groups"]`;
+    expect(snippetGroupsSelector).toHaveCount(3);
+    expect(queryAllTexts(snippetGroupsSelector)).toEqual(["A", "B", "C"]);
+    const imgSrc = queryAll(`${snippetGroupsSelector} img`).map((img) => img.dataset.src);
+    expect(imgSrc).toEqual(["a.svg", "b.svg", "c.svg"]);
+});
+
+test("display inner content snippet", async () => {
+    await setupWebsiteBuilder("<div><p>Text</p></div>", {
+        snippets: {
+            snippet_content: [
+                `<div name="Button A" data-oe-thumbnail="buttonA.svg" data-oe-snippet-id="123">
+                    <a class="btn btn-primary" href="#" data-snippet="s_button">Button A</a>
+                </div>`,
+                `<div name="Button B" data-oe-thumbnail="buttonB.svg" data-oe-snippet-id="123">
+                    <a class="btn btn-primary" href="#" data-snippet="s_button">Button B</a>
+                </div>`,
+            ],
+        },
+    });
+    await openSnippetsMenu();
+    const snippetInnerContentSelector = `.o-snippets-menu [data-category="snippet_content"]`;
+    expect(snippetInnerContentSelector).toHaveCount(2);
+    expect(queryAllTexts(snippetInnerContentSelector)).toEqual(["Button A", "Button B"]);
+    const imgSrc = queryAll(`${snippetInnerContentSelector} img`).map((img) => img.dataset.src);
+    expect(imgSrc).toEqual(["buttonA.svg", "buttonB.svg"]);
+});
+
+test("open add snippet dialog + switch snippet category", async () => {
+    const snippetsDescription = [
+        {
+            name: "Test",
+            groupName: "a",
+            content: getBasicSection("Yop"),
+        },
+        {
+            name: "Test",
+            groupName: "a",
+            content: getBasicSection("Hello"),
+        },
+        {
+            name: "Test",
+            groupName: "b",
+            content: getBasicSection("Nice"),
+        },
+    ];
+
+    await setupWebsiteBuilder("<div><p>Text</p></div>", {
+        snippets: {
+            snippet_groups: [
+                '<div name="A" data-oe-thumbnail="a.svg" data-oe-snippet-id="123" data-o-snippet-group="a"><section data-snippet="s_snippet_group"></section></div>',
+                '<div name="B" data-oe-thumbnail="b.svg" data-oe-snippet-id="123" data-o-snippet-group="b"><section data-snippet="s_snippet_group"></section></div>',
+            ],
+            snippet_structure: snippetsDescription.map((snippetDesc) =>
+                getSnippetStructure(snippetDesc)
+            ),
+        },
+    });
+    await openSnippetsMenu();
+    const snippetGroupsSelector = `.o-snippets-menu [data-category="snippet_groups"]`;
+    expect(queryAllTexts(snippetGroupsSelector)).toEqual(["A", "B"]);
```

Comment:
More a question here but maybe you can remove this `expect` as this is the goal of the `display group snippet` test?